### PR TITLE
Updater uses cUrlClient

### PR DESCRIPTION
**Pull request by NiLSPACE**

### CODE REVIEW

File: Classes/Updater.lua [hunk 1]
```diff
@@ -13,127 +13,63 @@ cUpdater = {}
...
-	OnReceivedData = function(a_TCPLink, a_Data)
-		-- Append the received Data to the saved data
-		g_WorldEditZip = g_WorldEditZip .. a_Data
-	end,
+function cUpdater:CheckForNewerVersion()
+	if (g_CheckAttempts > g_Config.Updates.NumAttempts) then
+		LOGWARNING("Could not connect to Github to check for a newer version for WorldEdit")
+		return;
+	end
+	g_CheckAttempts = g_CheckAttempts + 1;
```
Comment (madmaxoft) [inline Classes/Updater.lua]: How about storing this in the `cUpdater` itself? Access via `self.m_CheckAttempts`

Comment (NiLSPACE) [inline Classes/Updater.lua]: Good idea, I'll do that when I get home.

**Comment by NiLSPACE** [comment]
Done

**Comment by NiLSPACE** [comment]
Merge?

File: Classes/Updater.lua [hunk 2]
```diff
@@ -11,129 +11,65 @@ cUpdater = {}
...
-				g_DownloadAttempts = g_DownloadAttempts + 1
-				cUpdater:DownloadLatestVersion()
+	cUrlClient:Get("https://raw.githubusercontent.com/cuberite/WorldEdit/master/Info.lua",
+		function(a_Body, a_Data)
+			if (a_Body) then
+				
+				cUpdater:ParsePluginInfo(a_Body)
 			else
-				LOGWARNING("Error while downloading newer worldedit version")
+				cUpdater:CheckForNewerVersion()
```
Comment (madmaxoft) [inline Classes/Updater.lua]: Perhaps add a comment - `The request failed, retry`. You could also schedule a task onto the default world to retry later.

File: Classes/Updater.lua [hunk 3]
```diff
@@ -11,129 +11,65 @@ cUpdater = {}
...
+function cUpdater:DownloadLatestVersion(a_DisplayVersion)
+	if (cUpdater.s_DownloadAttempts > g_Config.Updates.NumAttempts) then
+		LOGWARNING("Error while downloading newer worldedit version")
+		return
+	end
+	cUpdater.s_DownloadAttempts = cUpdater.s_DownloadAttempts + 1;
+	
+	cUrlClient:Get("https://raw.githubusercontent.com/cuberite/WorldEdit/zip/master", 
+		function (a_Body, a_Data)
+			if (a_Body:len() ~= tonumber(a_Data["Content-Length"])) then
```
Comment (madmaxoft) [inline Classes/Updater.lua]: This will fail if the request fails - a_Body will be `nil` in such a case.

File: Classes/Updater.lua (hunk 2 above)
Comment (NiLSPACE) [inline Classes/Updater.lua]: Perhaps we need cuberite/cuberite#1754 for that?

**Comment by NiLSPACE** [comment]
Done

File: Classes/Updater.lua [hunk 4]
```diff
@@ -11,129 +11,74 @@ cUpdater = {}
...
+function cUpdater:DownloadLatestVersion(a_DisplayVersion)
+	if (cUpdater.s_DownloadAttempts > g_Config.Updates.NumAttempts) then
+		LOGWARNING("Error while downloading newer worldedit version")
+		return
+	end
+	cUpdater.s_DownloadAttempts = cUpdater.s_DownloadAttempts + 1;
+	
+	cUrlClient:Get("https://raw.githubusercontent.com/cuberite/WorldEdit/zip/master", 
+		function (a_Body, a_Data)
+			if (not a_Body or (a_Body:len() ~= tonumber(a_Data["Content-Length"]))) then
```
Comment (NiLSPACE) [inline Classes/Updater.lua]: @madmaxoft I was wondering, does cUrlClient check if the `content-length` is correct? And if not, should it?

Comment (madmaxoft) [inline Classes/Updater.lua]: It should work - the content-length is used (if the response isn't chunked) to distinguish the response's end and the potential beginning of another response over the same connection (we don't use that yet, though), so if the server cuts the answer short, the parser should report an error.

Comment (NiLSPACE) [inline Classes/Updater.lua]: Shall I remove the check here then?

**Comment by NiLSPACE** [comment]
Merge?

**Comment by madmaxoft** [comment]
I think it's good.

**Pull request merged**